Library/Homebrew/utils.rb

```diff
@@ -337,6 +337,14 @@ def with_system_path
   ENV["PATH"] = old_path
 end
 
+def with_custom_locale(locale)
+  old_locale = ENV["LC_ALL"]
+  ENV["LC_ALL"] = locale
+  yield
+ensure
+  ENV["LC_ALL"] = old_locale
+end
```

Comment:
How do you feel about generalizing this method a bit? Making it accept arbitrary changes to the environment would be simple, in fact it would only mean stealing [this code from `testing_env.rb`](https://github.com/Homebrew/brew/blob/7a00f03c92607f65336be0285584e8351ad8a73a/Library/Homebrew/test/testing_env.rb#L128-L136). That way we wouldn't have to come up with a special-purpose method for every case where we need to temporarily alter the environment. On the call site, this would be used as:

```ruby
with_environment("LC_ALL" => "C") { … }
```

Just an idea. Feel free to ignore it, particularly if others also think generalizing this wouldn't be as useful.